### ISSUE TITLE
Can a SARIF be converted outside of Github?

### ISSUE THREAD

**Issue by hector-vera** [oasis-tcs/sarif-spec#502]

**Comment by eddynaka** [comment]
Hi @hector-vera , if you use dotnet there is the sarif multitool.
we also have the npm sarif multitool.

What are you trying to accomplish?

Thank you!

**Comment by hector-vera** [comment]
Thanks so much for the quick reply.  It isn’t me.  A customer was asking to convert a static scan into SARIF format outside of Github.   I was only aware of this being done in Github.  I Googled and found a site but it seems to all evolve Github.  Nice but customer is seeking alternatives…..

Thanks!

- H.

--
Hector Escarcega
***@***.******@***.***>
480.430.1542<tel:%20480.430.1542> m
[signature_81893556]<https://www.veracode.com/>



From: Eddy Nakamura ***@***.***>
Reply-To: oasis-tcs/sarif-spec ***@***.***>
Date: Tuesday, November 16, 2021 at 8:25 PM
To: oasis-tcs/sarif-spec ***@***.***>
Cc: Hector Escarcega ***@***.***>, Mention ***@***.***>
Subject: [EXTERNAL] Re: [oasis-tcs/sarif-spec] Can a SARIF be converted outside of Github? (Issue #502)


This email originated from outside of Veracode.

________________________________
Hi @hector-vera<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fhector-vera&data=04%7C01%7Chescarcega%40veracode.com%7C0da343b10ce744c362da08d9a98239d4%7C3b627b68f21c4ed79fe3698efdedbe21%7C0%7C0%7C637727199047230499%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=ad9hvFY9Dxvb9YBu8uI2KQ2ScSHwpW5UK3VcD4GdLPs%3D&reserved=0> , if you use dotnet there is the sarif multitool.
we also have the npm sarif multitool.

What are you trying to accomplish?

Thank you!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Foasis-tcs%2Fsarif-spec%2Fissues%2F502%23issuecomment-970453918&data=04%7C01%7Chescarcega%40veracode.com%7C0da343b10ce744c362da08d9a98239d4%7C3b627b68f21c4ed79fe3698efdedbe21%7C0%7C0%7C637727199047240454%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=tdURMGB2ZiDEpWbV%2B7AUmaFLbExX0QexhSyhyMTEqdU%3D&reserved=0>, or unsubscribe<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAVM374M5USQXOZRTLAXXOTLUMKCLNANCNFSM5IEW6WAA&data=04%7C01%7Chescarcega%40veracode.com%7C0da343b10ce744c362da08d9a98239d4%7C3b627b68f21c4ed79fe3698efdedbe21%7C0%7C0%7C637727199047240454%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=mLj5J4kmwXkzlg2xHI4grysrFQh81mLNeuvH7kGY%2B9I%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Chescarcega%40veracode.com%7C0da343b10ce744c362da08d9a98239d4%7C3b627b68f21c4ed79fe3698efdedbe21%7C0%7C0%7C637727199047250417%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=97NNp%2F%2B%2F%2B0VzCrrBQWJwckIu21UhgAwWVWFHQPeuy4I%3D&reserved=0> or Android<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Chescarcega%40veracode.com%7C0da343b10ce744c362da08d9a98239d4%7C3b627b68f21c4ed79fe3698efdedbe21%7C0%7C0%7C637727199047250417%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=zXEixuDaIa1vSb%2Ft1oan11yYQ9zHqBkkPglYZ8MBzvE%3D&reserved=0>.

**Comment by Jeeppler** [comment]
@hector-vera yes, you can convert SARIF outside of GitHub. SARIF is not limited to GitHub. However, the product which creates the report either a) has to support SARIF or b) a converter converts to SARIF. A good example for a) is GoSec: https://github.com/securego/gosec. GoSec can output SARIF reports directly. A good example for b) is Bandit: https://github.com/PyCQA/bandit. Bandit v1.70 and earlier cannot output the report in SARIF directly, instead there is a third-party plugin which one has to install to get a SARIF report: https://pypi.org/project/bandit-sarif-formatter/.
